### PR TITLE
feat: add JWT authentication (without nonce)

### LINKED ISSUE #8: Implement JWT authentication (without nonce)
- JWT auth with secp256k1
  - without nonce for now
- Use JWT auth for REST APIs which were already implemented

### PR DESCRIPTION
**Pull request by youngjoon-lee**
Closes #8 

For unit tests with mocking, I changed `QueryClient` as interface.

### CODE REVIEW

File: server/middleware/context.go [hunk 1]
```diff
@@ -0,0 +1,3 @@
+package middleware
+
+type ContextKeyAuthenticatedAccountAddress struct{}
```
Comment (youngjoon-lee) [inline server/middleware/context.go:3]: If you're not familiar with using `struct{}` for context keys, please see https://velog.io/@kimmachinegun/Go-context.WithValue-%EC%95%88%EC%A0%84%ED%95%98%EA%B2%8C-%EC%82%AC%EC%9A%A9%ED%95%98%EA%B8%B0

**Review by gyuguen — APPROVED** [review]
Good

**Review by H4NLee — APPROVED** [review]
:+1::+1::+1:

**Review by inchori — APPROVED** [review]
LGTM

**Pull request merged**